src/utils/clmfates_interfaceMod.F90

```diff
@@ -1095,30 +1138,45 @@ subroutine wrap_update_hlmfates_dyn(this, nc, bounds_clump,      &
           ! protext it here with a lower bound of 0.0_r8.
 
           patch%wt_ed(col%patchi(c)) = max(0.0_r8, &
+
                1.0_r8-sum(this%fates(nc)%bc_out(s)%canopy_fraction_pa(1:npatch)))
 
-          if(sum(this%fates(nc)%bc_out(s)%canopy_fraction_pa(1:npatch))>1.0_r8)then
-             write(iulog,*)'Projected Canopy Area of all FATES patches'
-             write(iulog,*)'cannot exceed 1.0'
-             !end_run()
-          end if
+          patch%sp_pftorder_index(col%patchi(c)) = 0 !bg is the 0th patch in the SP FATES structure
 
-          do ifp = 1, npatch
+          areacheck = patch%wt_ed(col%patchi(c)) ! this is where we start the areachecking
+
+          do ifp = 1, this%fates(nc)%sites(s)%youngest_patch%patchno
+             ! for the vegetated patches
 
              p = ifp+col%patchi(c)
 
              ! bc_out(s)%canopy_fraction_pa(ifp) is the area fraction
-             ! the site's total ground area that is occupied by the 
-             ! area footprint of the current patch's vegetation canopy 
+             ! the site's total ground area that is occupied by the
+             ! area footprint of the current patch's vegetation canopy
 
              patch%is_veg(p) = .true.
              patch%wt_ed(p)  = this%fates(nc)%bc_out(s)%canopy_fraction_pa(ifp)
+             areacheck = areacheck + patch%wt_ed(p)
+
              elai(p) = this%fates(nc)%bc_out(s)%elai_pa(ifp)
-             tlai(p) = this%fates(nc)%bc_out(s)%tlai_pa(ifp)
              esai(p) = this%fates(nc)%bc_out(s)%esai_pa(ifp)
-             tsai(p) = this%fates(nc)%bc_out(s)%tsai_pa(ifp)
              hbot(p) = this%fates(nc)%bc_out(s)%hbot_pa(ifp)
-             htop(p) = this%fates(nc)%bc_out(s)%htop_pa(ifp)
+
+             if(use_fates_sp)then
+               canopystate_inst%tlai_hist_patch = this%fates(nc)%bc_out(s)%tlai_pa(ifp)
```

Comment:
Should this and the following two lines be indexed by p as in the following?
```suggestion
               canopystate_inst%tlai_hist_patch(p) = this%fates(nc)%bc_out(s)%tlai_pa(ifp)
```

Reply:
Good catch!  This definitely improved the TLAI output comparison: https://github.com/glemieux/fates-jupyter/blob/main/spmode/spmode-baseline-gencomp.ipynb